### PR TITLE
Register login

### PR DESCRIPTION
**Pull request by mkorob**
- user page
- log out button on all pages except lobby
- warning message when you log out
- clickable rows in leaderboard (and color fill on hover)
- the leaderbord vs. create account buttons

### CODE REVIEW

File: src/components/ui/Header.js [hunk 1]
```diff
@@ -1,30 +1,65 @@
...
 import PropTypes from "prop-types";
 import "styles/ui/Header.scss";
+import LogoutIcon from '@mui/icons-material/Logout';
+import { defaultTheme } from "../../styles/themes/defaulTheme";
+import { ThemeProvider } from "@emotion/react";
+import CustomPopUp from "../ui/CustomPopUp";
+import { Button } from 'components/ui/Button';
 
-const username = localStorage.getItem("username") ? localStorage.getItem("username") : null;
+const isRegistered = true;
```
Comment (apple00juice) [inline src/components/ui/Header.js:11]: this always true

Comment (mkorob) [reply src/components/ui/Header.js:11]: sorry ya changed it when testing the props thing and forgot... thanks!!

File: src/components/ui/BaseContainer.js [hunk 2]
```diff
@@ -5,15 +5,18 @@ import Header from "./Header";
...
-        <div {...props} className={`base-container ${props.className ?? ''}`}>
-            <Header/>
+        <div className={`base-container ${props.className ?? ''}`}>
+            <Header
+             logOutBool = {props.logOutBool}
+            />
             {props.children}
         </div>
     </div>
 );
```
Comment (apple00juice) [inline src/components/ui/BaseContainer.js:16]: You can add
`
BaseContainer.defaultProps = {
    logoutBool: false
};
`

that by default it is set to false and not null (I think the default would be null or unassigned). The same for Header, just changing BaseContainer to Header.
Then you just have to add it where it is true.

File: src/components/views/lobby/ShareLobbyCode.js [hunk 3]
```diff
@@ -99,7 +99,7 @@ const ShareLobbyCode = ({ id }) => {
     }
 
     return (
-        <BaseContainer>
+        <BaseContainer lobbyBool = {false}>
```
Comment (apple00juice) [inline src/components/views/lobby/ShareLobbyCode.js:102]: This should be logOutBool right

**Review by apple00juice — APPROVED** [review]
Rest looks good, I think you have to add logOutBool = {true} to the Header in HomePage because we don't have a BaseContainer there

File: src/components/views/lobby/Lobby.js [hunk 4]
```diff
@@ -242,7 +242,7 @@ const Lobby = ({ id }) => {
     }, [id, token]);
 
     return (
-        <BaseContainer>
+        <BaseContainer logOutBool = {true}>
```
[the remark this answers is not in the document]
Comment (mkorob) [reply src/components/views/lobby/Lobby.js:245]: I guess technically the variable should be called the noLogOutBool..

File: src/components/ui/BaseContainer.js (hunk 2 above)
Comment (apple00juice) [reply src/components/ui/BaseContainer.js:16]: also true

File: src/components/views/lobby/Lobby.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (apple00juice) [reply src/components/views/lobby/Lobby.js:245]: Ahh, I see then it is correct.

**Review by apple00juice — APPROVED** [review]
Sorry for nagging, but there is noLogOutBool = {true} missing in updateLobby, invite-users and share/qr.
Rest looks good though! 👍

**Review by luis-tm — APPROVED** [review]

**Pull request merged**